itest/rust/src/engine_tests/gfile_test.rs

```diff
@@ -0,0 +1,137 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::io::{BufRead, Read, Seek, SeekFrom, Write};
+
+use crate::framework::itest;
+use godot::builtin::GString;
+use godot::engine::{file_access::ModeFlags, GFile};
+
+const TEST_FULL_PATH: &str = "res://file_tests";
+
+fn remove_test_file() {
+    let test_file_path = std::path::Path::new(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/../godot/",
+        "file_tests"
+    ));
+    std::fs::remove_file(test_file_path)
+        .unwrap_or_else(|_| panic!("Couldn't remove test file: {}", test_file_path.display()));
+}
+
+#[itest]
+fn non_unique_error_works() {
+    use godot::engine::NotUniqueError;
+    use godot::engine::RefCounted;
+
+    let unique = RefCounted::new();
+    assert!(NotUniqueError::check(unique).is_ok());
+
+    let shared = RefCounted::new();
+    let cloned = shared.clone();
+    assert!(NotUniqueError::check(shared).is_err());
+    if let Err(error) = NotUniqueError::check(cloned) {
+        assert_eq!(error.get_reference_count(), 2);
+    }
+}
+
+#[itest]
+fn basic_read_write_works() {
+    let mut file = GFile::open(TEST_FULL_PATH, ModeFlags::WRITE).unwrap();
+    let line_to_store = GString::from("TESTING1");
+    file.write_gstring_line(line_to_store.clone()).unwrap();
+    drop(file);
+    let mut file = GFile::open(TEST_FULL_PATH, ModeFlags::READ).unwrap();
+    let gotten_line = file.read_gstring_line().unwrap();
+    assert_eq!(line_to_store, gotten_line);
+    drop(file);
+    remove_test_file();
```

Comment:
Group the related tests by adding empty lines:
```suggestion
    let mut file = GFile::open(TEST_FULL_PATH, ModeFlags::WRITE).unwrap();
    let line_to_store = GString::from("TESTING1");
    file.write_gstring_line(line_to_store.clone()).unwrap();
    drop(file);
    
    let mut file = GFile::open(TEST_FULL_PATH, ModeFlags::READ).unwrap();
    let gotten_line = file.read_gstring_line().unwrap();
    assert_eq!(line_to_store, gotten_line);
    drop(file);
    
    remove_test_file();
```